python/ql/src/semmle/python/objects/ObjectAPI.qll

```diff
@@ -639,6 +639,28 @@ class BuiltinMethodValue extends FunctionValue {
     override int maxParameters() { none() }
 }
 
+/**
+ * Class representing bound-methods, such as `o.func`, where `o` is an instance
+ * of a class that has a callable attribute `func`.
+ *
+ * This is currently NOT a CallableValue, see explanation in https://github.com/Semmle/ql/pull/3182
+ */
+class BoundMethodValue extends Value {
+    BoundMethodValue() { this instanceof BoundMethodObjectInternal }
+
+    /**
+     * Gets the callable that will be used when `this` called.
+     * The actual callable for `func` in `o.func`.
+     */
+    CallableValue getFunction() { result = this.(BoundMethodObjectInternal).getFunction() }
+
+    /**
+     * Gets the value that will be used for the 'self' parameter when `this` is called.
```

Comment:
Should this be `self` instead of 'self'?
If not, what's the difference?

Reply:
Very true :+1: (as defined in https://github.com/Semmle/ql/blob/master/docs/ql-style-guide.md#documentation _Documentation for specific items_ point 9)